### DISCUSSION TITLE
Air App Not Loading Stylesheets and Javascript

### DISCUSSION THREAD

**Discussion by slendertech** [airsdk/Adobe-Runtime-Support#3307] [Q&A]
So, I'm writing an air application that uses html as opposed to swf and actionscript. Everything was smooth sailing until I wanted to add css and include the AIRAliases.js file. They are linked to index.html but no matter what I do to change their paths around, the air app will never use them. When I inspect the webpage within the air application, there are 0 console errors and the styles in the stylesheets do not exist on any elements. Additionally, the css and js files are not seen in the sources tab.

Does anyone have any idea what the heck is happening here? Am I getting the paths wrong? Where should I place these files? From what I read online, they go in the `src` folder. I've also tried putting them in the root folder and in the `bin` folder with no avail.

```html
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8"/>
	<title></title>
	<script src="src/js/AIRAliases.js" type="text/javascript"></script>
	<link rel="stylesheet" type="text/css" href="src/css/cssnormalize.min.css"> 
</head>
<body>
	<p>Hello World!</p>
</body>
</html>
```

P.S. Setting the stylesheet href to a web address (https://blahblahblah/blah.css) that hosts a stylesheet externally does seem to work. I would rather not have to do that though.

**Comment by ajwfrost** [comment]
Can you confirm what versions you're using for the AIR SDK, and if on Windows, whether you're using the new Edge WebView2 or the older IE engine? Plus knowing what platform it's running on would also help us look at reproducing things.

I sounds like something is going wrong in the paths; I would have thought they needed to be in the application folder ('bin' I guess) because they'd be resolved at run-time, but I have a vague feeling we'd seen that local paths didn't work for some webview implementations. Which would imply you'd need to load them from an internal webserver mechanism.. unless we'd be able to override the handling of resource loading.

thanks

**Reply by slendertech** [reply]
Yes, I am using the WebView version on Windows. I was using FlashDevelop though, so I thought that might have been the issue. I'm trying out IntellijIDEA and seeing how it goes.

**Reply by slendertech** [reply]
Okay, well now intellij idea won't display the html file in the air app despite <content> being set to main.html in the app descriptor. Instead, when I build the project, it replaces `main.html` in the app descriptor template I made `with _the name of my app_.swf`. Any idea what's happening?

**Reply by slendertech** [reply]
Okay, so fixed above problem. But IDEA is now giving me the original problem stated in the question title. Please help!

**Reply by ajwfrost** [reply]
We've been trying a few things here .. on Windows in the Edge WebView2, we are meant to be able to pick up on resource requests and redirect them, but it doesn't appear to be working for relative addresses that are provided when the app is loaded from file. If we modify your HTML code to e.g. use an "app:/js/someting.js" URL, then we do get the request and can (presumably) redirect that to load it from the application folder.

So will need to check a little more to see if we can get it to work like that automatically...

Just to double-check, are you setting the HTML filename in your application descriptor directly? Rather than loading via any ActionScript class such as StageWebView?

thanks

**Reply by ajwfrost** [reply]
Hi @slendertech  - in case this is a scenario where you have `<content>main.html</content>` within your application descriptor file, can you please try using the DLL from the below link?
https://transfer.harman.com/message/JHAAM8HZnjFikCz7KRakgu

I just realised I didn't add the paths for where this file is in an existing SDK, for it to be overwritten (please back up the existing files before overwriting, so that you can restore it later!)
runtimes\air\win\Adobe AIR\Versions\1.0
runtimes\air-captive\win\Adobe AIR\Versions\1.0

thanks

**Reply by slendertech** [reply]
I'm sorry, I actually just decided to use electron instead. But I will try it when I have the chance. Though, I won't be available all through this week.

**Reply by ajwfrost** [reply]
I had wondered - if you were just hosting an HTML-only application - whether AIR was the best choice here!
But, I am hoping that the change we've made for the above will then make things better. The other useful thing would be an internal webserver so that you could have things running locally via http rather than as files.. :-)

thanks
